ee/tabby-webserver/src/repositories/resolve.rs

```diff
@@ -257,24 +211,23 @@ impl RepositoryCache {
         Ok(resp.map(boxed))
     }
 
-    pub fn resolve_meta(&self, key: &RepositoryKey) -> Option<RepositoryMeta> {
-        if let Some(meta) = self.repositories().get(key) {
-            return Some(meta.clone());
-        }
-        None
+    pub fn resolve_meta(&self, key: &RepositoryKey) -> Result<RepositoryMeta> {
```

Comment:
you can translate the repository key from `(name, path)` to `(git_url, path)` to avoid storing repository file.